### PR TITLE
Porting to Qgis 3 

### PR DESCRIPTION
**Pull request by jorgealmerio**
Hi, Olivier

Sorry for the delay.

Here are the files changed for porting the plugin to the new python 3.xx format (Qgis 3)

I realized that "vectorbendertransformers.py" from here (master) is outdated comparing to the Qgis repository version, that's why there were so many code differences between my updated version and the master.

Thank you!

### CODE REVIEW

**Comment by jorgealmerio** [comment]
Hi Olivier,

I did the all the uploads directly from github website. As I told you I am not github expert, I do not know exactly how to solve this line ending issues, but I saw a way to you see the real changes to the "vectorbenderdialog.py" by clicking on my second commint (I do not know why there is 3), please see pictures:
vectorbenderdialog

image

File: vectorbender.py [hunk 1]
```diff
@@ -103,28 +105,25 @@ def determineTransformationType(self):
             0 if no pairs Found
             1 if one pair found => translation
             2 if two pairs found => linear
-            3 if two pairs found => linear
```
Comment (olivierdalang) [inline vectorbender.py]: I think this was mislabeled in the original code (should have been `3 if three pairs found => affine, 4 if four or more pairs found => bending`), but must be kept for affine transformation to still work.

File: vectorbender.py [hunk 2]
```diff
@@ -103,28 +105,25 @@ def determineTransformationType(self):
...
             return 0
 
-        featuresCount = len(pairsLayer.selectedFeaturesIds()) if self.dlg.restrictBox_pairsLayer.isChecked() else len(pairsLayer.allFeatureIds())
+        featuresCount = len(pairsLayer.selectedFeatureIds()) if self.dlg.restrictBox_pairsLayer.isChecked() else len(pairsLayer.allFeatureIds())
         
         if featuresCount == 1:
             return 1
         elif featuresCount == 2:
             return 2
-        elif featuresCount == 3:
```
Comment (olivierdalang) [inline vectorbender.py]: same as above, we must keep the functionality for affine transform

File: vectorbender.py [hunk 3]
```diff
@@ -139,13 +138,10 @@ def run(self):
 
         # Loading the delaunay
         restrictToSelection = self.dlg.restrictBox_pairsLayer.isChecked()
-        if transType==4:
+        if transType==3:
             self.dlg.displayMsg( "Loading delaunay mesh (%i points) ..." % len(self.ptsA) )
             QCoreApplication.processEvents()
             self.transformer = BendTransformer( pairsLayer, restrictToSelection, self.dlg.bufferValue() )
-        elif transType==3:
```
Comment (olivierdalang) [inline vectorbender.py]: same as above...

File: vectorbendertransformers.py [hunk 4]
```diff
@@ -111,25 +111,19 @@ def __init__(self, pairsLayer, restrictToSelection):
 
 
         """
-
```
Comment (olivierdalang) [inline vectorbendertransformers.py]: Revert these unrelated/unnecessary changes (for the whole file).

If you wish to improve the style, that's welcome, but should be part of another PR (so that this one really stays about porting to QGIS3)

File: vectorbendertransformers.py [hunk 5]
```diff
@@ -175,8 +166,8 @@ def __init__(self, pairsLayer, restrictToSelection):
 
     def map(self, p):
 
-        return QgsPoint( self.a*p.x()+self.b*p.y()+self.c, self.d*p.x()+self.e*p.y()+self.f )
-
+        return QgsPointXY( self.a*p.x()+self.b*p.y()+self.c, self.d*p.x()+self.e*p.y()+self.f ) #Almerio: era QgsPoint
```
Comment (olivierdalang) [inline vectorbendertransformers.py]: Remove these unnecessary `#Almerio: era QgsPoint` comments  (for the whole file)

File: metadata.txt [hunk 6]
```diff
@@ -23,8 +24,9 @@ email=olivier.dalang@gmail.com
 
 # Uncomment the following line and add your changelog entries:
 changelog=- 2014-05-22 - Version 0.0 : intial release
-- 2014-05-26 - Version 0.1 : linear and uniform transformation method, scipy dependency no more needed, better management of pinned points
```
Comment (olivierdalang) [inline metadata.txt]: Is this reformatting necessary for Qgis 3 compatibility ? If not, please revert.

If you wish to improve the style, that's welcome, but should be part of another PR (so that this one really stays about porting to QGIS3)

File: vectorbender.py (hunk 1 above)
Comment (jorgealmerio) [reply vectorbender.py]: Done!

File: vectorbender.py (hunk 2 above)
Comment (jorgealmerio) [reply vectorbender.py]: Done!

File: vectorbender.py (hunk 3 above)
Comment (jorgealmerio) [reply vectorbender.py]: Done!

File: vectorbendertransformers.py (hunk 4 above)
Comment (jorgealmerio) [reply vectorbendertransformers.py]: I tried to revert, I don't know if it's good now, because of number of spaces characters.

File: vectorbendertransformers.py (hunk 5 above)
Comment (jorgealmerio) [reply vectorbendertransformers.py]: Done!

**Comment by jorgealmerio** [comment]
1. Exception when running a transform (any type)
Line 269:
newGeom = QgsGeometry.fromPolyline( [geom[-1],geom[-1]] )
replaced by:
newGeom = QgsGeometry.fromPolylineXY( [geom[-1],geom[-1]] )

2. Dialog doesn't properly refresh
On vectorbenderdialog.py
Line 267:
if event.type() == QEvent.FocusIn:
replaced by:
if event.type() == QEvent.WindowActivate:

3. Deprecation warning (on plugin loading)
Line 152 from "objcreator.py" from Qgis is using a deprecated option it should be changed to:
plugin = open(filename, 'r')

It happens with some plugins (I am not so sure, by it depends on the way it calls the UI files), so it's not a Vectorbender warning.

**Comment by olivierdalang** [comment]
Thanks @jorgealmerio , quickly tested today and it looks fine ! I'll push the new version to the repo ASAP
